src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
```

Comment:
1. Use .reduce() instead of forEach() again
2. As you can see this code repeats twice, so it'll be good if we will create an additional function-helper getAverageAge() which will accept an array with values and return an average age of that array.
```suggestion
  return women.reduce((prev, cur) => prev + cur.died - cur.born, 0)/women.length;
```